File: ts/src/distributions/distribution.ts
/**
 * Base interface for distributions that can generate samples.
 */

import { Rng } from '../rng';

/**
 * Base interface for distributions that can generate samples.
 */
export interface Distribution {
  /**
   * Generate a single sample from this distribution.
   */
  sample(rng: Rng): number;

  /**
   * Generate multiple samples from this distribution.
   */
  samples(rng: Rng, count: number): number[];
}

File: kt/src/main/kotlin/dev/pragmastat/distributions/Distribution.kt
package dev.pragmastat.distributions

import dev.pragmastat.Rng

/**
 * Base interface for distributions that can generate samples.
 */
interface Distribution {
    /**
     * Generate a single sample from this distribution.
     */
    fun sample(rng: Rng): Double

    /**
     * Generate multiple samples from this distribution.
     */
    fun samples(rng: Rng, count: Int): List<Double> {
        return (0 until count).map { sample(rng) }
    }
}
